pkg/sources/reconciler/azureservicebustopicsource/subscription.go

```diff
@@ -280,23 +195,16 @@ func recursErrMsg(errMsg string, err error) string {
 // isNotFound returns whether the given error indicates that some Azure
 // resource was not found.
 func isNotFound(err error) bool {
-	if dErr := (autorest.DetailedError{}); errors.As(err, &dErr) {
-		return dErr.StatusCode == http.StatusNotFound
-	}
-	return false
+	return errors.Is(err, servicebussdk.ErrNotFound{})
 }
 
 // isDenied returns whether the given error indicates that a request to the
 // Azure API could not be authorized.
 // This category of issues is unrecoverable without user intervention.
 func isDenied(err error) bool {
-	if dErr := (autorest.DetailedError{}); errors.As(err, &dErr) {
-		if code, ok := dErr.StatusCode.(int); ok {
-			return code == http.StatusUnauthorized || code == http.StatusForbidden
-		}
-	}
-
-	return false
+	// FIXME(antoineco): the SDK returns unstructured errors
+	// request failed: 401 SubCode=40100: Unauthorized : Unauthorized access for 'DeleteSubscription' operation on endpoint 'sb://triggermesh-dev.servicebus.windows.net/triggermesh-dev/subscriptions/io.triggermesh.azureservicebussources-4197625275?api-version=2017-04'. Tracking Id: bbbccdfa-9277-407f-891a-18a59fa82c7c_G20
+	return strings.Contains(err.Error(), "Unauthorized access")
```

Comment:
Another occurrence terrible error handling based on string parsing.
This time for "Unauthorized" errors, which our logic expects to be handled properly.